### ISSUE TITLE
Incorrect Subversion information on Fedora

### ISSUE THREAD

**Issue by duboism** [rohieb/ctan-support-vc#1]
Hello,

I just discovered the vc bundle and tried it with a project versioned with `Subversion`. Unfortunately, it doesn't work out of the box: the information in the `vc.tex` file is incorrect (it says the wc is at revision 0 which is not true, the repository information is missing and so on).

Digging a bit, I think I understood the problem:

1. on my Fedora box, the `LC_ALL` environment variable doesn't exist so the `LC_ALL=C` line doesn't work as expected: it creates a bash variable but not an environment variable so the `svn` commands run in my current locale (`fr_FR.UTF-8`) and so the `vc-svn.awk` script doesn't work
1. the project on which I tested `vc` contains non-C characters (`é` and so on) in this case, using `LC_ALL=C` doesn't work:
```
$ LC_ALL=C svn info
svn: E000022: Can't convert string from native encoding to 'UTF-8':
svn: E000022: </path/to/my/wc-with-é>
```

The fix for 1 is not hard: use `export LC_ALL=C`.

I think I have a fix for 2 but I'm not sure if it will work on all platforms:
```
$ LC_ALL=C.UTF-8 svn info
<Expected output of `svn info`>
```

Any thoughts ?

**Comment by duboism** [comment]
A quick look at the `vc` script for other SCM shows that they use the same technique so may be affected by the same problems.

**Comment by duboism** [comment]
[This SO answer](https://unix.stackexchange.com/questions/597962/how-widespread-is-the-c-utf-8-locale) suggests that `C.UTF-8` is not widely accepted but might soon be. So I guess that we can fix problem 1 and just report an error for problem 2 for now.

**Comment by rohieb** [comment]
> [This SO answer](https://unix.stackexchange.com/questions/597962/how-widespread-is-the-c-utf-8-locale) suggests that `C.UTF-8` is not widely accepted but might soon be. So I guess that we can fix problem 1 and just report an error for problem 2 for now.

Uh… I guess waiting for glibc 2.35 to roll out will certainly help things, but I'd like to find a better solution for the meantime. I don't have access to any SVN repos, do you know whether SVN has some command or option to export machine-readable data about the repo or the current commit?

**Comment by duboism** [comment]
>  I don't have access to any SVN repos, do you know whether SVN has some command or option to export machine-readable data about the repo or the current commit?

Good point. Most `svn` subcommands (including the ones used by `vc`, `info` and `status`) have the `--xml` option for that. The schema are available in [`svn` repo ](https://svn.apache.org/repos/asf/subversion/trunk/subversion/svn/schema/). Parsing this with `awk` might be a bit difficult though.

Note that the problem with non-ASCII characters remains.

**Comment by duboism** [comment]
By the way if you need to hack with a subversion repository, you can simply do:
```bash
$ svnadmin create /tmp/my-repo  # Get wiped at each reboot
$ cd /path/to/my/wc
$ svn co file:///tmp/my-repo
```

**Comment by duboism** [comment]
Just to mention that [glibc 2.35](https://sourceware.org/pipermail/libc-alpha/2022-February/136040.html) has been released with support for `C.UTF-8`.
